backend/session/direct-freebsd.c

```diff
@@ -139,6 +139,50 @@ static int vt_handler(int signo, void *data) {
 	return 1;
 }
 
+static bool get_tty_path(int tty, char path[static 11], size_t len) {
+	assert(tty > 0);
+
+	const char prefix[] = "/dev/ttyv";
+	static_assert(sizeof(prefix) + 1 <= 11, "TTY path prefix is too long");
+
+	const size_t prefix_len = sizeof(prefix) - 1;
+	strcpy(path, prefix);
+
+	size_t offset = prefix_len;
+	int num = tty - 1;
+	if (num == 0) {
+		path[offset++] = '0';
+		path[offset++] = '\0';
+		return true;
+	}
+
+	const int base = 32;
+	size_t num_len = 0;
+	for (int remaning = num; remaning > 0; remaning /= base, num_len++);
```

Comment:
Instead we can just set this to a reasonable value.

Reply:
Fixed by moving the length check into the loop.